src/model/MazeCompression.java

```diff
@@ -0,0 +1,269 @@
+package model;
+
+import java.awt.*;
+import java.io.*;
+import java.util.*;
+
+
+// Huffman tree node:
+class Node {
+    Character ch;
+    Integer freq;
+    Node left = null;
+    Node right = null;
+
+    Node(Character ch, Integer freq) {
+        this.ch = ch;
+        this.freq = freq;
+    }
+
+    public Node(Character ch, Integer freq, Node left, Node right) {
+        this.ch = ch;
+        this.freq = freq;
+        this.left = left;
+        this.right = right;
+    }
+
+    public Node() {}
+}
+
+public class MazeCompression {
+
+    public static int _mazeCnt;
+    File _mazeCntFile = new File("mazeCnt.bin");
+
+    public MazeCompression() throws IOException {
+
+        // if mazeCnt.txt exists:
+        if (_mazeCntFile.exists() && !_mazeCntFile.isDirectory()) {
+
+            // read the existing mazeCnt:
+            DataInputStream dis = new DataInputStream(new FileInputStream(_mazeCntFile));
+            _mazeCnt = dis.read();
+            dis.close();
+        }
+
+        // if mazeCnt.txt not exists:
+        else {
+            // create the file and set the val to 0:
+            boolean mazeCntTxtCreated = _mazeCntFile.createNewFile();
+            if (!mazeCntTxtCreated)
+                throw new IOException("Unable to create mazeCnt file at specified path. Please check that!");
+
+            DataOutputStream dos = new DataOutputStream(new FileOutputStream(_mazeCntFile));
+            dos.write(0);
+            dos.close();
+        }
+    }
+
+    public int getUpdateMazeCnt() throws IOException {
```

Comment:
Maybe short exp about this funcs meaning?

Reply:
done